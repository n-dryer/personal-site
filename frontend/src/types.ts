import { LucideIcon } from 'lucide-react';

export interface Project {
  name: string;
  description: string;
}

export interface Experience {
  id: string;
  title: string;
  company: string;
  location: string;
  date: string;
  description: string;
  icon: LucideIcon;
  achievements: string[];
  technologies: string[];
  projects: Project[];
}

export interface Skill {
  id: string;
  name: string;
  level: number;
  icon?: string;
}

export interface SocialLink {
  name: string;
  url: string;
}

export interface UserData {
  fullName: string;
  bioLine: string;
  photoUrl: string;
  email: string;
  phone: string;
  location: string;
  socialLinks: SocialLink[];
  resumeUrl: string;
}

// Legacy interfaces to maintain compatibility during migration
export interface LegacyExperience {
  title: string;
  company: string;
  location: string;
  dates: string;
  blurb: string[];
}
