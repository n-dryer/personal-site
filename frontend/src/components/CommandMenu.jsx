import React, { useState, useEffect, useRef } from 'react';
import { Command } from 'cmdk';
import { Download, Mail, Linkedin, Github, Moon, Sun, ChevronRight } from 'lucide-react';

const CommandMenu = ({ userData, toggleTheme, darkMode }) => {
  const [open, setOpen] = useState(false);
  const [inputValue, setInputValue] = useState('');
  const inputRef = useRef(null);
  const listRef = useRef(null);

  // Define command items
  const items = [
    { 
      label: "Download résumé", 
      icon: Download, 
      href: "/nathan_dryer_resume.pdf", 
      shortcut: "⏎", 
      primary: true 
    },
    { 
      label: "Email Nate", 
      icon: Mail, 
      href: `mailto:${userData.email}` 
    },
    { 
      label: "LinkedIn", 
      icon: Linkedin, 
      href: userData.socialLinks.find(link => link.name === "LinkedIn")?.url || "#" 
    },
    { 
      label: "GitHub", 
      icon: Github, 
      href: userData.socialLinks.find(link => link.name === "GitHub")?.url || "#" 
    },
    { 
      label: `Toggle ${darkMode ? 'light' : 'dark'} mode`, 
      icon: darkMode ? Sun : Moon, 
      action: "theme" 
    },
  ];

  // Filter items based on search input
  const filteredItems = inputValue === '' 
    ? items 
    : items.filter(item => 
        item.label.toLowerCase().includes(inputValue.toLowerCase())
      );

  // Handle keyboard shortcuts
  useEffect(() => {
    const handleKeyDown = (e) => {
      // Open command palette with Cmd+K or Ctrl+K
      if ((e.metaKey || e.ctrlKey) && e.key === 'k') {
        e.preventDefault();
        setOpen(true);
      }
    };

    window.addEventListener('keydown', handleKeyDown);
    return () => window.removeEventListener('keydown', handleKeyDown);
  }, []);

  // Auto-focus input on palette open only on desktop
  useEffect(() => {
    if (open && window.innerWidth >= 768) {
      setTimeout(() => {
        inputRef.current?.focus();
      }, 100);
    }
  }, [open]);

  // Handle email copy
  const handleEmailCopy = () => {
    navigator.clipboard.writeText(userData.email);
    // You could add a toast notification here
    setOpen(false);
  };

  return (
    <Command
      label="Site command palette"
      className="w-full max-w-full md:max-w-md mx-auto rounded-2xl bg-white/65 dark:bg-slate-900/50
                backdrop-blur-sm shadow-lg ring-1 ring-black/5 dark:ring-white/10
                focus-within:ring-blue-500 dark:focus-within:ring-blue-400 transition"
    >
      <Command.Input
        ref={inputRef}
        value={inputValue}
        onValueChange={setInputValue}
        placeholder="Type a command…"
        className="h-12 w-full bg-transparent px-4 text-base text-gray-800 dark:text-white outline-none"
        autoComplete="off"
      />
      <Command.List 
        ref={listRef}
        className="grid gap-1 p-2"
      >
        {filteredItems.map(({ label, icon: Icon, href, shortcut, primary, action }) => (
          <Command.Item 
            key={label} 
            className="rounded-xl focus:bg-slate-100 dark:focus:bg-white/5 outline-none"
            role="menuitem"
          >
            {href ? (
              <a
                href={href}
                className={`flex items-center gap-3 px-4 py-3 rounded-xl w-full min-h-[48px]
                          ${primary 
                            ? "bg-gradient-to-r from-blue-500 to-blue-600 text-white hover:from-blue-600 hover:to-blue-700" 
                            : "hover:bg-slate-100 dark:hover:bg-white/5"}`}
                download={href.includes('resume') ? "Nathan_Dryer_Resume.pdf" : undefined}
              >
                <div className={`flex items-center justify-center w-8 h-8 ${primary 
                  ? "bg-white/20 rounded-full" 
                  : "text-blue-500 dark:text-blue-400"}`}>
                  <Icon className="w-5 h-5 shrink-0" />
                </div>
                <span className="grow text-left">{label}</span>
                {shortcut && (
                  <kbd className="px-2 py-1 text-xs bg-black/5 dark:bg-white/10 rounded">
                    {shortcut}
                  </kbd>
                )}
                {!primary && !shortcut && <ChevronRight className="w-4 h-4 opacity-50" />}
              </a>
            ) : (
              <button
                onClick={() => {
                  if (action === "theme") {
                    toggleTheme();
                  }
                  setOpen(false);
                }}
                className="flex items-center gap-3 px-4 py-3 rounded-xl w-full min-h-[48px]
                          hover:bg-slate-100 dark:hover:bg-white/5"
              >
                <div className="flex items-center justify-center w-8 h-8 text-blue-500 dark:text-blue-400">
                  <Icon className="w-5 h-5 shrink-0" />
                </div>
                <span className="grow text-left text-gray-800 dark:text-white">{label}</span>
              </button>
            )}
          </Command.Item>
        ))}
      </Command.List>
    </Command>
  );
};

export default CommandMenu;
