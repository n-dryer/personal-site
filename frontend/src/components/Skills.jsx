import React from 'react';
import { motion } from 'framer-motion';
import { useInView } from 'react-intersection-observer';

const Skills = ({ userData }) => {
  const [ref, inView] = useInView({
    threshold: 0.1,
    triggerOnce: false,
  });

  // Animation variants
  const containerVariants = {
    hidden: { opacity: 0 },
    visible: {
      opacity: 1,
      transition: {
        staggerChildren: 0.1,
      },
    },
  };

  const itemVariants = {
    hidden: { opacity: 0, y: 20 },
    visible: {
      opacity: 1,
      y: 0,
      transition: {
        type: 'spring',
        stiffness: 50,
        damping: 10,
      },
    },
  };

  return (
    <section id="skills" ref={ref} className="py-20 bg-white dark:bg-dark-800">
      <div className="container mx-auto px-6">
        <motion.div 
          className="text-center mb-16"
          initial={{ opacity: 0, y: 20 }}
          animate={inView ? { opacity: 1, y: 0 } : { opacity: 0, y: 20 }}
          transition={{ duration: 0.5 }}
        >
          <h2 className="text-3xl font-display font-bold mb-4 text-gray-900 dark:text-white">Expertise & Skills</h2>
          <p className="text-base text-gray-600 dark:text-gray-300 max-w-2xl mx-auto">
            I specialize in building AI/ML products with a focus on exceptional user experience.
          </p>
        </motion.div>

        <motion.div
          className="grid grid-cols-2 md:grid-cols-4 gap-4 md:gap-6"
          variants={containerVariants}
          initial="hidden"
          animate={inView ? "visible" : "hidden"}
        >
          {userData.skills.map((skill, index) => (
            <motion.div
              key={index}
              className="skill-card"
              variants={itemVariants}
              whileHover={{ y: -5 }}
              whileTap={{ scale: 0.98 }}
            >
              <div className="skill-icon">
                <div className="skill-icon-inner"></div>
              </div>
              <h3 className="skill-name">{skill.name}</h3>
            </motion.div>
          ))}
        </motion.div>
      </div>
    </section>
  );
};

export default Skills;
