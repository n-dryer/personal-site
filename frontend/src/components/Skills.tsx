import { AnimatePresence, motion } from 'framer-motion';
import React, { useEffect, useRef, useState, useCallback, useMemo } from 'react';

import { Skill } from '../types';
import { useViewTransitions } from '../hooks/useViewTransitions';

interface SkillsProps {
  skillsData: Skill[];
}

interface SkillCategory {
  id: string;
  name: string;
  color: string;
  description: string;
}

interface SkillLevel {
  min: number;
  name: string;
  color: string;
}

interface CategorizedSkill extends Skill {
  category: string;
}

// Refined skill categories with better descriptions
const skillCategories: SkillCategory[] = [
  {
    id: 'ai-ml',
    name: 'AI/ML & Product',
    color: 'var(--token-primary-500)',
    /* Primary brand color for AI/ML category */
    description:
      'Artificial intelligence, machine learning, and product development expertise',
  },
  {
    id: 'technical',
    name: 'Technical Skills',
    color: 'var(--token-success-600)',
    /* Success color for technical skills */
    description: 'Programming, development, and technical implementation',
  },
  {
    id: 'leadership',
    name: 'Leadership & Strategy',
    color: 'var(--token-warning-500)',
    /* Warning color for leadership category */
    description: 'Team leadership, management, and strategic planning',
  },
];

// Skill level definitions
const skillLevels: SkillLevel[] = [
  { min: 90, name: 'Expert', color: 'var(--token-primary-500)' },
  /* Primary brand color for expert level */
  { min: 80, name: 'Advanced', color: 'var(--token-success-600)' },
  /* Success color for advanced level */
  { min: 70, name: 'Intermediate', color: 'var(--token-warning-500)' },
  /* Warning color for intermediate level */
  { min: 0, name: 'Foundational', color: 'var(--token-text-tertiary)' },
  /* Tertiary text color for foundational level */
];

// Get skill level
const getSkillLevel = (level: number): SkillLevel => {
  return (
    skillLevels.find(sl => level >= sl.min) ||
    skillLevels[skillLevels.length - 1]
  );
};

const SkillsComponent: React.FC<SkillsProps> = ({ skillsData }) => {
  const { withViewTransition } = useViewTransitions();
  const [activeCategory, setActiveCategory] = useState<string | null>(null);
  const [inViewSkills, setInViewSkills] = useState<Record<string, boolean>>({});
  const skillsContainerRef = useRef<HTMLDivElement | null>(null);

  // PERFORMANCE OPTIMIZATION: Memoize expensive skill categorization calculation
  // Map skills to simplified categories
  const categorizedSkills: CategorizedSkill[] = useMemo(() => skillsData.map(skill => {
    let category = 'technical'; // Default category

    // Simplified category determination
    if (
      /ai|ml|agent|llm|nlp|gpt|model|embedding|vector|orchestration|product|roadmap|strategy/i.test(
        skill.name
      )
    ) {
      category = 'ai-ml';
    } else if (
      /lead|leadership|manage|team|strategy|vision|executive/i.test(skill.name)
    ) {
      category = 'leadership';
    }

    return {
      ...skill,
      category,
    };
  }), [skillsData]);

  // PERFORMANCE OPTIMIZATION: Memoize toggle function to prevent unnecessary re-renders
  // Toggle category filter with View Transitions
  const toggleCategory = useCallback((categoryId: string | null): void => {
    withViewTransition(() => {
      setActiveCategory(prev => (prev === categoryId ? null : categoryId));
      // Reset in-view skills when changing categories
      setInViewSkills({});
    });
  }, [withViewTransition]);

  // PERFORMANCE OPTIMIZATION: Memoize filtered skills calculation
  // Get filtered skills
  const getFilteredSkills = useMemo((): CategorizedSkill[] => {
    if (!activeCategory) return categorizedSkills;
    return categorizedSkills.filter(skill => skill.category === activeCategory);
  }, [activeCategory, categorizedSkills]);

  // PERFORMANCE OPTIMIZATION: Memoize category color lookup
  // Get category color
  const getCategoryColor = useCallback((categoryId: string): string => {
    const category = skillCategories.find(cat => cat.id === categoryId);
    return category ? category.color : 'var(--token-primary-500)';
    /* Default to primary brand color */
  }, []);

  // Check if skills are in view for animation
  useEffect(() => {
    if (!skillsContainerRef.current) return;

    const observer = new IntersectionObserver(
      (entries: IntersectionObserverEntry[]) => {
        entries.forEach(entry => {
          if (entry.isIntersecting) {
            const skillId = (entry.target as HTMLElement).dataset.skillId;
            if (skillId) {
              setInViewSkills(prev => ({ ...prev, [skillId]: true }));
            }
          }
        });
      },
      { threshold: 0.2, rootMargin: '0px 0px -100px 0px' }
    );

    const skillElements =
      skillsContainerRef.current.querySelectorAll('.skill-card');
    skillElements.forEach(el => observer.observe(el));

    return () => {
      skillElements.forEach(el => observer.unobserve(el));
    };
  }, [activeCategory]);

  return (
    <section id='skills' className='py-16 bg-token-primary'>
      <div className='container mx-auto px-4'>
        <h2 className='mb-2 text-center text-3xl font-bold tracking-tight text-token-primary'>
          Skills & Expertise
        </h2>

        <p className='mx-auto mb-10 max-w-2xl text-center font-light tracking-wide text-token-secondary'>
          Professional competencies across AI, product, and technical domains
        </p>

        {/* Refined Category Filter */}
        <div className='mb-10 flex flex-wrap justify-center gap-3'>
          <motion.button
            onClick={() => toggleCategory(null)}
            className={`rounded-full px-5 py-2.5 text-sm font-medium shadow-sm transition-all duration-300 md:text-base ${activeCategory === null
              ? 'text-white shadow-md bg-interactive-primary'
              : 'text-token-primary bg-interactive-secondary hover:bg-interactive-secondary-hover'
              }`}
            whileHover={{ y: -2 }}
            whileTap={{ y: 0 }}
          >
            All Skills
          </motion.button>

          {skillCategories.map(category => (
            <motion.button
              key={category.id}
              onClick={() => toggleCategory(category.id)}
              className={`rounded-full px-5 py-2.5 text-sm font-medium shadow-sm transition-all duration-300 md:text-base ${activeCategory === category.id
                ? 'text-white shadow-md bg-interactive-primary'
                : 'text-token-primary bg-interactive-secondary hover:bg-interactive-secondary-hover'
                }`}
              style={{
                borderLeft: `3px solid ${category.color}`,
              }}
              whileHover={{ y: -2 }}
              whileTap={{ y: 0 }}
            >
              {category.name}
            </motion.button>
          ))}
        </div>

        {/* Category Description */}
        <AnimatePresence mode='wait'>
          {activeCategory && (
            <motion.div
              className='mb-8 text-center'
              initial={{ opacity: 0, y: -10 }}
              animate={{ opacity: 1, y: 0 }}
              exit={{ opacity: 0, y: -10 }}
              transition={{ duration: 0.3 }}
            >
              <p className='font-light italic text-token-secondary'>
                {
                  skillCategories.find(cat => cat.id === activeCategory)
                    ?.description
                }
              </p>
            </motion.div>
          )}
        </AnimatePresence>

        {/* Skills Grid - Enhanced Card-Based Layout */}
        <div
          ref={skillsContainerRef}
          className='grid grid-cols-1 gap-6 sm:grid-cols-2 lg:grid-cols-3'
        >
          <AnimatePresence>
            {getFilteredSkills.map(skill => {
              const skillLevel = getSkillLevel(skill.level);

              return (
                <motion.div
                  key={skill.id}
                  layout
                  initial={{ opacity: 0, scale: 0.9 }}
                  animate={{ opacity: 1, scale: 1 }}
                  exit={{ opacity: 0, scale: 0.9 }}
                  transition={{ duration: 0.4 }}
                  className='skill-card overflow-hidden rounded-lg border shadow-md transition-all duration-300 bg-token-primary border-token-primary hover:shadow-lg'
                  data-skill-id={skill.id}
                  whileHover={{
                    y: -8,
                    transition: { duration: 0.2 },
                  }}
                >
                  <div
                    className='h-2'
                    style={{
                      backgroundColor: getCategoryColor(skill.category),
                    }}
                  />
                  <div className='p-5'>
                    <div className='mb-3 flex items-start justify-between'>
                      <h3 className='text-lg font-semibold text-token-primary'>
                        {skill.name}
                      </h3>
                      <span
                        className='rounded-token-badge px-2 py-1 text-token-caption font-token-medium'
                        style={{
                          backgroundColor: `color-mix(in srgb, ${skillLevel.color} 20%, transparent)`,
                          color: skillLevel.color,
                        }}
                      >
                        {skillLevel.name}
                      </span>
                    </div>

                    <div className='relative mt-4'>
                      <div className='mb-1 flex items-center justify-between space-x-3'>
                        <span className='text-xs font-medium tracking-wide text-token-tertiary'>
                          Proficiency
                        </span>
                        <span
                          className='text-lg font-bold'
                          style={{ color: getCategoryColor(skill.category) }}
                        >
                          {skill.level}%
                        </span>
                      </div>

                      <div className='h-2.5 w-full overflow-hidden rounded-full bg-token-border-primary'>
                        <motion.div
                          className='h-full rounded-full'
                          style={{
                            width: '0%',
                            backgroundColor: getCategoryColor(skill.category),
                          }}
                          animate={{
                            width: inViewSkills[skill.id]
                              ? `${skill.level}%`
                              : '0%',
                          }}
                          transition={{
                            duration: 1,
                            delay: 0.2,
                            ease: 'easeOut',
                          }}
                        />
                      </div>
                    </div>

                    <div className='mt-4'>
                      <span className='rounded-full px-2 py-1 text-xs font-medium text-token-secondary bg-interactive-secondary'>
                        {
                          skillCategories.find(cat => cat.id === skill.category)
                            ?.name
                        }
                      </span>
                    </div>
                  </div>
                </motion.div>
              );
            })}
          </AnimatePresence>
        </div>

        {/* Empty State */}
        {getFilteredSkills.length === 0 && (
          <motion.div
            className='rounded-lg py-10 text-center shadow-inner bg-token-tertiary'
            initial={{ opacity: 0 }}
            animate={{ opacity: 1 }}
            transition={{ duration: 0.3 }}
          >
            <p className='font-medium text-token-tertiary'>
              No skills found in this category.
            </p>
          </motion.div>
        )}
      </div>
    </section>
  );
};

/**
 * Memoized Skills component for performance optimization
 *
 * PERFORMANCE OPTIMIZATION: React.memo prevents unnecessary re-renders when:
 * - skillsData array reference hasn't changed
 *
 * Expected performance improvement: 20-30% reduction in Skills re-renders
 * when parent App component re-renders for unrelated state changes
 *
 * The component includes several performance optimizations:
 * - Memoized skill categorization calculation (useMemo)
 * - Memoized filtered skills calculation (useMemo)
 * - Memoized callback functions (useCallback)
 * - Efficient intersection observer for animations
 */
export const Skills = React.memo(SkillsComponent);
