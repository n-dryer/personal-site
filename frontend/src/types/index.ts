// Re-export from portfolio.ts for backward compatibility
export * from './portfolio';
