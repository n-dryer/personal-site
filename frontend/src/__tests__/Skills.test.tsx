import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import { Skills } from '../components/Skills';
import { Skill } from '../types';

// Mock framer-motion
jest.mock('framer-motion', () => ({
  motion: {
    div: ({ children, ...props }: React.PropsWithChildren<any>) => (
      <div data-testid="motion-div" {...props}>{children}</div>
    ),
    button: ({ children, ...props }: React.PropsWithChildren<any>) => (
      <button data-testid="motion-button" {...props}>{children}</button>
    ),
  },
  AnimatePresence: ({ children, mode }: React.PropsWithChildren<any>) => <div data-testid="animate-presence">{children}</div>,
}));

// Mock IntersectionObserver
const mockIntersectionObserver = jest.fn();
mockIntersectionObserver.mockReturnValue({
  observe: jest.fn(),
  unobserve: jest.fn(),
  disconnect: jest.fn(),
});
window.IntersectionObserver = mockIntersectionObserver;

describe('Skills Component', () => {
  // Create minimal mock data
  const mockSkillsData: Skill[] = [
    { 
      id: 'skill-1', 
      name: "Test Skill 1", 
      level: 95 
    },
    { 
      id: 'skill-2', 
      name: "Test Skill 2", 
      level: 85 
    },
    { 
      id: 'skill-3', 
      name: "Test Skill 3", 
      level: 75 
    },
  ];

  it('renders correctly', () => {
    const { container } = render(<Skills skillsData={mockSkillsData} />);
    expect(container).toMatchSnapshot();
  });

  it('displays all skills initially', () => {
    render(<Skills skillsData={mockSkillsData} />);
    expect(screen.getByText('Test Skill 1')).toBeInTheDocument();
    expect(screen.getByText('Test Skill 2')).toBeInTheDocument();
    expect(screen.getByText('Test Skill 3')).toBeInTheDocument();
  });

  it('filters skills when a category is selected', () => {
    render(<Skills skillsData={mockSkillsData} />);
    
    // Find and click the "AI/ML & Product" category button
    const categoryButtons = screen.getAllByRole('button');
    const aiMlButton = categoryButtons.find(button => button.textContent === 'AI/ML & Product');
    
    if (aiMlButton) {
      fireEvent.click(aiMlButton);
      
      // Check that the category description is displayed
      expect(screen.getByText(/artificial intelligence, machine learning/i)).toBeInTheDocument();
    }
  });

  it('resets to all skills when "All Skills" is clicked', () => {
    render(<Skills skillsData={mockSkillsData} />);
    
    // First click a category filter
    const categoryButtons = screen.getAllByRole('button');
    const aiMlButton = categoryButtons.find(button => button.textContent === 'AI/ML & Product');
    
    if (aiMlButton) {
      fireEvent.click(aiMlButton);
      
      // Then click "All Skills" to reset
      const allSkillsButton = screen.getByText('All Skills');
      fireEvent.click(allSkillsButton);
      
      // All skills should be visible again
      expect(screen.getByText('Test Skill 1')).toBeInTheDocument();
      expect(screen.getByText('Test Skill 2')).toBeInTheDocument();
      expect(screen.getByText('Test Skill 3')).toBeInTheDocument();
    }
  });
});
