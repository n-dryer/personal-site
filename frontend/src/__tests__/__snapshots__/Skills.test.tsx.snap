// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills Component renders correctly 1`] = `
<div>
  <section
    class="py-16 bg-white dark:bg-gray-800"
    id="skills"
  >
    <div
      class="container mx-auto px-4"
    >
      <h2
        class="text-3xl font-bold text-center mb-2 text-gray-800 dark:text-white tracking-tight"
      >
        Skills & Expertise
      </h2>
      <p
        class="text-center text-gray-600 dark:text-gray-300 mb-10 max-w-2xl mx-auto font-light tracking-wide"
      >
        Professional competencies across AI, product, and technical domains
      </p>
      <div
        class="flex flex-wrap justify-center gap-3 mb-10"
      >
        <button
          class="px-5 py-2.5 rounded-full text-sm md:text-base font-medium transition-all duration-300 shadow-sm bg-blue-600 text-white shadow-md"
          data-testid="motion-button"
          whilehover="[object Object]"
          whiletap="[object Object]"
        >
          All Skills
        </button>
        <button
          class="px-5 py-2.5 rounded-full text-sm md:text-base font-medium transition-all duration-300 shadow-sm bg-gray-100 text-gray-700 dark:bg-gray-700 dark:text-gray-200 hover:bg-gray-200 dark:hover:bg-gray-600"
          data-testid="motion-button"
          style="border-left: 3px solid #3B82F6;"
          whilehover="[object Object]"
          whiletap="[object Object]"
        >
          AI/ML & Product
        </button>
        <button
          class="px-5 py-2.5 rounded-full text-sm md:text-base font-medium transition-all duration-300 shadow-sm bg-gray-100 text-gray-700 dark:bg-gray-700 dark:text-gray-200 hover:bg-gray-200 dark:hover:bg-gray-600"
          data-testid="motion-button"
          style="border-left: 3px solid #10B981;"
          whilehover="[object Object]"
          whiletap="[object Object]"
        >
          Technical Skills
        </button>
        <button
          class="px-5 py-2.5 rounded-full text-sm md:text-base font-medium transition-all duration-300 shadow-sm bg-gray-100 text-gray-700 dark:bg-gray-700 dark:text-gray-200 hover:bg-gray-200 dark:hover:bg-gray-600"
          data-testid="motion-button"
          style="border-left: 3px solid #F59E0B;"
          whilehover="[object Object]"
          whiletap="[object Object]"
        >
          Leadership & Strategy
        </button>
      </div>
      <div
        data-testid="animate-presence"
      />
      <div
        class="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-3 gap-6"
      >
        <div
          data-testid="animate-presence"
        >
          <div
            animate="[object Object]"
            class="skill-card bg-white dark:bg-gray-800 rounded-lg shadow-md overflow-hidden border border-gray-200 dark:border-gray-700 hover:shadow-lg transition-all duration-300"
            data-skill-id="skill-1"
            data-testid="motion-div"
            exit="[object Object]"
            initial="[object Object]"
            transition="[object Object]"
            whilehover="[object Object]"
          >
            <div
              class="h-2"
              style="background-color: rgb(16, 185, 129);"
            />
            <div
              class="p-5"
            >
              <div
                class="flex justify-between items-start mb-3"
              >
                <h3
                  class="font-semibold text-lg text-gray-800 dark:text-white"
                >
                  Test Skill 1
                </h3>
                <span
                  class="text-xs px-2 py-1 rounded-full font-medium"
                  style="background-color: rgba(59, 130, 246, 0.125); color: rgb(59, 130, 246);"
                >
                  Expert
                </span>
              </div>
              <div
                class="mt-4 relative"
              >
                <div
                  class="flex justify-between items-center mb-1"
                >
                  <span
                    class="text-xs text-gray-500 dark:text-gray-400 font-medium tracking-wide"
                  >
                    Proficiency
                  </span>
                  <span
                    class="text-lg font-bold"
                    style="color: rgb(16, 185, 129);"
                  >
                    95
                    %
                  </span>
                </div>
                <div
                  class="h-2.5 w-full rounded-full bg-gray-200 dark:bg-gray-700 overflow-hidden"
                >
                  <div
                    animate="[object Object]"
                    class="h-full rounded-full"
                    data-testid="motion-div"
                    style="width: 0%; background-color: rgb(16, 185, 129);"
                    transition="[object Object]"
                  />
                </div>
              </div>
              <div
                class="mt-4"
              >
                <span
                  class="text-xs px-2 py-1 rounded-full bg-gray-100 dark:bg-gray-700 text-gray-600 dark:text-gray-300 font-medium"
                >
                  Technical Skills
                </span>
              </div>
            </div>
          </div>
          <div
            animate="[object Object]"
            class="skill-card bg-white dark:bg-gray-800 rounded-lg shadow-md overflow-hidden border border-gray-200 dark:border-gray-700 hover:shadow-lg transition-all duration-300"
            data-skill-id="skill-2"
            data-testid="motion-div"
            exit="[object Object]"
            initial="[object Object]"
            transition="[object Object]"
            whilehover="[object Object]"
          >
            <div
              class="h-2"
              style="background-color: rgb(16, 185, 129);"
            />
            <div
              class="p-5"
            >
              <div
                class="flex justify-between items-start mb-3"
              >
                <h3
                  class="font-semibold text-lg text-gray-800 dark:text-white"
                >
                  Test Skill 2
                </h3>
                <span
                  class="text-xs px-2 py-1 rounded-full font-medium"
                  style="background-color: rgba(16, 185, 129, 0.125); color: rgb(16, 185, 129);"
                >
                  Advanced
                </span>
              </div>
              <div
                class="mt-4 relative"
              >
                <div
                  class="flex justify-between items-center mb-1"
                >
                  <span
                    class="text-xs text-gray-500 dark:text-gray-400 font-medium tracking-wide"
                  >
                    Proficiency
                  </span>
                  <span
                    class="text-lg font-bold"
                    style="color: rgb(16, 185, 129);"
                  >
                    85
                    %
                  </span>
                </div>
                <div
                  class="h-2.5 w-full rounded-full bg-gray-200 dark:bg-gray-700 overflow-hidden"
                >
                  <div
                    animate="[object Object]"
                    class="h-full rounded-full"
                    data-testid="motion-div"
                    style="width: 0%; background-color: rgb(16, 185, 129);"
                    transition="[object Object]"
                  />
                </div>
              </div>
              <div
                class="mt-4"
              >
                <span
                  class="text-xs px-2 py-1 rounded-full bg-gray-100 dark:bg-gray-700 text-gray-600 dark:text-gray-300 font-medium"
                >
                  Technical Skills
                </span>
              </div>
            </div>
          </div>
          <div
            animate="[object Object]"
            class="skill-card bg-white dark:bg-gray-800 rounded-lg shadow-md overflow-hidden border border-gray-200 dark:border-gray-700 hover:shadow-lg transition-all duration-300"
            data-skill-id="skill-3"
            data-testid="motion-div"
            exit="[object Object]"
            initial="[object Object]"
            transition="[object Object]"
            whilehover="[object Object]"
          >
            <div
              class="h-2"
              style="background-color: rgb(16, 185, 129);"
            />
            <div
              class="p-5"
            >
              <div
                class="flex justify-between items-start mb-3"
              >
                <h3
                  class="font-semibold text-lg text-gray-800 dark:text-white"
                >
                  Test Skill 3
                </h3>
                <span
                  class="text-xs px-2 py-1 rounded-full font-medium"
                  style="background-color: rgba(245, 158, 11, 0.125); color: rgb(245, 158, 11);"
                >
                  Intermediate
                </span>
              </div>
              <div
                class="mt-4 relative"
              >
                <div
                  class="flex justify-between items-center mb-1"
                >
                  <span
                    class="text-xs text-gray-500 dark:text-gray-400 font-medium tracking-wide"
                  >
                    Proficiency
                  </span>
                  <span
                    class="text-lg font-bold"
                    style="color: rgb(16, 185, 129);"
                  >
                    75
                    %
                  </span>
                </div>
                <div
                  class="h-2.5 w-full rounded-full bg-gray-200 dark:bg-gray-700 overflow-hidden"
                >
                  <div
                    animate="[object Object]"
                    class="h-full rounded-full"
                    data-testid="motion-div"
                    style="width: 0%; background-color: rgb(16, 185, 129);"
                    transition="[object Object]"
                  />
                </div>
              </div>
              <div
                class="mt-4"
              >
                <span
                  class="text-xs px-2 py-1 rounded-full bg-gray-100 dark:bg-gray-700 text-gray-600 dark:text-gray-300 font-medium"
                >
                  Technical Skills
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
